packages/eslint-plugin/src/rules/prefer-promise-reject-errors.ts

```diff
@@ -0,0 +1,147 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import { getDeclaredVariables } from '@typescript-eslint/utils/eslint-utils';
+import type * as ts from 'typescript';
+
+import {
+  createRule,
+  getParserServices,
+  isErrorLike,
+  isFunction,
+  isIdentifier,
+} from '../util';
+
+export type MessageIds = 'rejectAnError';
+
+export type Options = [
+  {
+    allowEmptyReject?: boolean;
+  },
+];
+
+export default createRule<Options, MessageIds>({
+  name: 'prefer-promise-reject-errors',
+  meta: {
+    type: 'suggestion',
+    docs: {
```

Comment:
What do you think about recommending this in the `strict-type-checked` config? I think it'd be reasonable to as we already include https://typescript-eslint.io/rules/no-throw-literal in there.

Reply:
Done https://github.com/typescript-eslint/typescript-eslint/pull/8011/commits/f2ee5d967496beda88be1e8c25feb317551a5097